src/interface/pipe.rs

```diff
@@ -94,8 +94,9 @@ impl EmulatedPipe {
         let mut read_end = self.read_end.lock().unwrap();
 
         while bytes_read < length {
-            if (read_end.len() == 0) & self.eof.load(Ordering::Relaxed) { break; }
-            let bytes_to_read = min(length, bytes_read + read_end.len());
+            let pipe_space = read_end.len();
+            if (pipe_space == 0) & self.eof.load(Ordering::Relaxed) { break; }
+            let bytes_to_read = min(length, bytes_read + pipe_space);
```

Comment:
This was *probably* compiler optimized out but it's good to have it this way if for nothing else except clarity. Looks good!

Reply:
It actually resulted in a significant speed increase, so afik it wasn't?